src/core/algorithms/cfd/cfd_discovery.h

```diff
@@ -26,8 +26,8 @@ class CFDDiscovery : public Algorithm {
 protected:
     config::InputTable input_table_;
 
-    unsigned columns_number_;
-    unsigned tuples_number_;
+    size_t columns_number_;
+    size_t tuples_number_;
```

Comment:
Why we use `size_t` for those fields? We have type aliases `model::ColumnIndex` and `model::TupleIndex` (they have type `unsigned int`)

Reply:
These are not indices, why not?

Reply:
I did this for easier overloading in `AddNewItemsInFullTable`. And these are not indices.

Reply:
It seems reasonable to me to use the same type for the number of columns as for the column index.
Because of this discrepancy in types, we have to write (line 55 in the `CreateFrom` function) a rather strange cast, which we probably would like to avoid:
```cpp
for (AttributeIndex i = 0; static_cast<size_t>(i) < num_columns; ++i) {
```
I'm sure it's better to change the type of the `num_columns` and `columns_number` to `AttributeIndex` to avoid this weird casting. I would prefer to fix this, but I don't insist.

Reply:
[May be related](https://github.com/Desbordante/desbordante-core/pull/346#discussion_r1564526352)
@Firsov62121

Reply:
I tried it, and it just leads to a bunch of other casts, not solving anything.